pkg/eks/create.go

```diff
@@ -463,3 +474,177 @@ func getParameterValueFromOutput(key string, outputs []*cloudformation.Output) s
 
 	return ""
 }
+
+// EnableEBSCSIDriverInput holds the options for enabling the EBS CSI driver
+type EnableEBSCSIDriverInput struct {
+	EKSService   services.EKSServiceInterface
+	IAMService   services.IAMServiceInterface
+	CFService    services.CloudFormationServiceInterface
+	Config       *eksv1.EKSClusterConfig
+	AddonVersion string
+}
+
+// EnableEBSCSIDriver manages the installation of the EBS CSI driver for EKS, including the
+// creation of the OIDC Provider, the IAM role and the validation and installation of the EKS add-on
+func EnableEBSCSIDriver(opts EnableEBSCSIDriverInput) error {
+	oidcID, err := configureOIDCProvider(opts.IAMService, opts.EKSService, opts.Config)
+	if err != nil {
+		return fmt.Errorf("could not configure oidc provider: %v", err)
+	}
+	roleArn, err := createEBSCSIDriverRole(opts.CFService, opts.Config, oidcID)
+	if err != nil {
+		return fmt.Errorf("could not create ebs csi driver role: %v", err)
+	}
+	_, err = installEBSAddon(opts.EKSService, opts.Config, roleArn, opts.AddonVersion)
+	if err != nil {
+		return fmt.Errorf("failed to install ebs csi driver addon: %v", err)
+	}
+
+	return nil
+}
+
+// CheckEBSAddon checks if the EBS CSI driver add-on is installed. If it is, it will return
+// the ARN of the add-on. If it is not, it will return an empty string. Otherwise, it will return an error
+func CheckEBSAddon(eksService services.EKSServiceInterface, config *eksv1.EKSClusterConfig) (string, error) {
+	input := eks.DescribeAddonInput{
+		AddonName:   aws.String(ebsCSIAddonName),
+		ClusterName: aws.String(config.Spec.DisplayName),
+	}
+
+	output, err := eksService.DescribeAddon(&input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == eks.ErrCodeResourceNotFoundException {
+				return "", nil
+			}
+		}
+		return "", err
+	}
+	if output.Addon == nil {
+		return "", nil
+	}
+
+	return *output.Addon.AddonArn, nil
+}
+
+func configureOIDCProvider(iamService services.IAMServiceInterface, eksService services.EKSServiceInterface, config *eksv1.EKSClusterConfig) (string, error) {
+	output, err := iamService.ListOIDCProviders(&iam.ListOpenIDConnectProvidersInput{})
+	if err != nil {
+		return "", err
+	}
+	clusterOutput, err := eksService.DescribeCluster(&eks.DescribeClusterInput{
+		Name: aws.String(config.Spec.DisplayName),
+	})
+	if err != nil {
+		return "", err
+	}
+	id := path.Base(*clusterOutput.Cluster.Identity.Oidc.Issuer)
```

Comment:
I know it's very unlikely to happen but can we check that `clusterOutput` is not nil?